### PR TITLE
feat(core): expand --from versions in migrate command to packages included in packageGroup

### PR DESCRIPTION
**Pull request by leosvelperez**
## Current Behavior

Running `nx migrate latest --from nx@15.0.0` only provides a version override for the `nx` package and not for the packages included in its `packageGroup`. Therefore, it collects package updates and migrations from the version provided only for the `nx` package.

## Expected Behavior

Running `nx migrate latest --from nx@15.0.0` should provide a version override for the `nx` package and the packages included in its `packageGroup`. It should collect package updates and migrations from the version provided for the `nx` package and the packages included in its `packageGroup`.

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/nx/src/command-line/migrate.ts [hunk 1]
```diff
@@ -434,6 +443,80 @@ export class Migrator {
...
+          this.installedPkgVersionOverrides[packageConfig.package] ??=
+            this.installedPkgVersionOverrides[packageName];
+        }
+      }
+
+      return typeof packageConfig === 'string'
+        ? { package: packageConfig, version: targetVersion }
+        : packageConfig;
+    });
+  }
```
Comment (AgentEnder) [inline packages/nx/src/command-line/migrate.ts:518]: I'm moving a lot of this logic into the readMigrationsConfig fn inside https://github.com/nrwl/nx/pull/14648, as a heads up. I'll approve + merge this and rebase 14648 to get it fixed up.

**Review by AgentEnder — APPROVED** [review]

File: packages/nx/src/command-line/migrate.ts [hunk 2]
```diff
@@ -434,6 +443,80 @@ export class Migrator {
...
+      };
+    }
+
+    if (!packageGroup) {
+      return [];
+    }
+
+    if (!Array.isArray(packageGroup)) {
+      return Object.entries(packageGroup).map(([pkg, version]) => {
+        if (this.installedPkgVersionOverrides[packageName] && version === '*') {
```
Comment (leosvelperez) [inline packages/nx/src/command-line/migrate.ts:495]: @FrozenPandaz I made the change so it applies to the `packageGroup` of any package, not just `nx`. Please let me know if we want to scope this change to only apply to the `nx` package (and maybe `@nrwl/workspace` for compat reasons).

Comment (AgentEnder) [reply packages/nx/src/command-line/migrate.ts:495]: `migrate` should work the same for community plugins that use package group, or Nx's package group. This is good :)

File: packages/nx/src/command-line/migrate.ts (hunk 1 above)
Comment (leosvelperez) [reply packages/nx/src/command-line/migrate.ts:518]: Cool, please let me know when you do to have a look and double-check things so nothing gets overlooked on the rebase.

File: packages/nx/src/command-line/migrate.ts (hunk 2 above)
Comment (leosvelperez) [reply packages/nx/src/command-line/migrate.ts:495]: yeah, I agree and that's why I implemented it that way, but wanted to be sure :)

**Pull request merged**